src/WebJobs.Extensions.DurableTask/BasicLogWriter.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+#if !FUNCTIONS_V1
+using Mono.Unix.Native;
+#endif
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// BasicLogWriter:
+    /// Simple file writer (which doesn't hold the handle open, with basic rollover).
+    /// </summary>
+    public class BasicLogWriter
+    {
+        private long fileSize;
+
+        /// <summary>
+        /// Simple log writer:
+        /// Rolls over at specified size (bytes)
+        /// Maximum of 2 files retained.
+        /// </summary>
+        /// <param name="logFolder">Directory where the logfile will be created.</param>
+        /// <param name="fileName">Log file name.</param>
+        /// <param name="rollOverSize">Maximum size of file (in bytes) before archiving it.</param>
+        public BasicLogWriter(string logFolder, string fileName, int rollOverSize)
+        {
+            this.LogFolder = logFolder;
+            this.RollOverSize = rollOverSize;
+            this.FileName = fileName;
+            this.Initialize();
+        }
+
+        internal long RollOverSize { get; private set; }
+
+        internal string LogFolder { get; private set; }
+
+        internal string FileName { get; private set; }
+
+        internal bool BufferingEnabled { get; set; }
+
+        private string LogFile
+        {
+            get
+            {
+                return Path.Combine(this.LogFolder, this.FileName);
+            }
+        }
+
+        private string LogFileBackup
+        {
+            get
+            {
+                return Path.Combine(this.LogFolder, this.FileName + ".1");
+            }
+        }
+
+        /// <summary>
+        /// Writes string argument to the log file.
+        /// </summary>
+        /// <param name="input">string to log.</param>
+        public void WriteLog(string input)
+        {
+            this.WriteDataInternal(input);
+        }
+
+        private void WriteDataInternal(string input)
+        {
+            try
+            {
+                // Roll-over if needed
+                if (this.fileSize >= this.RollOverSize)
+                {
+                    if (File.Exists(this.LogFileBackup))
+                    {
+                        try
+                        {
+                            File.Delete(this.LogFileBackup);
+                        }
+                        catch (Exception)
+                        {
+                        }
+                    }
+#if !FUNCTIONS_V1
+                    Syscall.rename(this.LogFile, this.LogFileBackup);
+                    this.fileSize = 0;
```

Comment:
While this works great for our purposes, it's weird to me that we have this `#if` condition here but no `#else` condition. Maybe just for good feelings we could add an `#else` that does a `File.Move(this.LogFile, this.LogFileBackup)`?

Reply:
Oh, maybe we're not using this anymore...

Reply:
We're not! I just pushed the port late last night so I didn't want to necessarily remove this at that point. We also still have the log4net dependency somewhere in here :p . I'll get to it shortly